crates/networking/p2p/rlpx/l2/messages.rs

```diff
@@ -0,0 +1,180 @@
+use crate::rlpx::{
+    message::{Message, RLPxMessage},
+    utils::{snappy_compress, snappy_decompress},
+};
+use bytes::BufMut;
+use ethrex_common::{
+    types::{batch::Batch, Block},
+    H256,
+};
+use ethrex_rlp::error::{RLPDecodeError, RLPEncodeError};
+use ethrex_rlp::structs::{Decoder, Encoder};
+use keccak_hash::keccak;
+use secp256k1::{Message as SecpMessage, SecretKey};
+use std::{ops::Deref as _, sync::Arc};
+
+#[derive(Debug, Clone)]
+pub struct NewBlock {
+    // Not ideal to have an Arc here, but without it, clippy complains
+    // that this struct is bigger than the other variant when used in the
+    // L2Message enum definition. Since we don't modify this
+    // block field, we don't need a Box, and we also get the benefit
+    // of (almost) freely cloning the pointer instead of the block iself
+    // when broadcasting this message.
+    pub block: Arc<Block>,
+    pub signature: [u8; 64],
+    pub recovery_id: [u8; 4],
+}
+
+impl RLPxMessage for NewBlock {
+    const CODE: u8 = 0x0;
+
+    fn encode(&self, buf: &mut dyn BufMut) -> Result<(), RLPEncodeError> {
+        let mut encoded_data = vec![];
+        Encoder::new(&mut encoded_data)
+            .encode_field(&self.block.deref().clone())
+            .encode_field(&self.signature)
+            .encode_field(&self.recovery_id)
+            .finish();
+        let msg_data = snappy_compress(encoded_data)?;
+        buf.put_slice(&msg_data);
+        Ok(())
+    }
+
+    fn decode(msg_data: &[u8]) -> Result<Self, RLPDecodeError> {
+        let decompressed_data = snappy_decompress(msg_data)?;
+        let decoder = Decoder::new(&decompressed_data)?;
+        let (block, decoder) = decoder.decode_field("block")?;
+        let (signature, decoder) = decoder.decode_field("signature")?;
+        let (recovery_id, decoder) = decoder.decode_field("recovery_id")?;
+        decoder.finish()?;
+        Ok(NewBlock {
+            block: Arc::new(block),
+            signature,
+            recovery_id,
+        })
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct BatchSealed {
+    pub batch: Batch,
```

Comment:
We can do the same thing here to solve clippy lints:
```suggestion
    pub batch: Arc<Batch>,
```